### ISSUE TITLE
Improve dealing with deleted / renamed files

### ISSUE THREAD

**Issue by runxel** [gingerbeardman/stapler#40]
If something gets renamed or removed (happens on network drives with colleagues all over it ;)) it would be nice to "recover" the latest path, or at least file name so you can then go and fix those missing documents by hand.

![grafik](https://github.com/user-attachments/assets/59bf4cb4-f3c5-4ab1-8081-89e659be82c2)

**Comment by gingerbeardman** [comment]
Good idea.

I will look to display the file name at least.

**Comment by runxel** [comment]
Another data point: Even without renaming Stapler has issues with files on a network share.
Could you look into this as well please? It looks like the screenshot above – even when the network share (locally run on a Synology NAS) is mounted and I can access the files just fine.

**Comment by gingerbeardman** [comment]
Stapler just stores the aliases/bookmarks, they are created and processed by macOS. 

They are not modified after they've added (hmm, should they be?). 

1. Open Stapler document in a plain text editor
1. Copy the bookmark encoded text
1. Unescape the text https://www.freeformatter.com/json-escape.html
1. Paste the Base64 encoded text into [Precize](https://eclecticlight.co/2024/08/12/whats-the-time-mr-apfs-with-a-new-version-of-precize/)

If be interested if we can spot the difference between a file that gets "lost" and if we can maybe there's something I can do about it.

**Comment by runxel** [comment]
Okay it got even weirder. Now Stapler freezes on opening the stapler file – and so does Precize!
Both do not recover and need to be killed...
Decoding the Base64 just "as-is" looks completely garbled, not sure if that is supposed to be like this:
![grafik](https://github.com/user-attachments/assets/1e109f62-343f-41e4-b811-51c242db60c4)

Also from what I can see non-ASCII chars are dropped, like the [German "ß"](https://en.wikipedia.org/wiki/%C3%9F) (sharp s).

**Comment by runxel** [comment]
I have another stapler also pointing to files on the same network share – that one works without a hitch. So it can't be the network share per se that is the issue.

**Comment by gingerbeardman** [comment]
This is what I see when I working with one of my bookmarks:

```
Ym9va7wCAAAAAAQQMAAAAHpF1pJqGLcWddHXLQvPkeOfGMmuf3ETMWg0pWwJoWO0uAEAAAQAAAADAwAAABgAKAUAAAABAQAAVXNlcnMAAAAEAAAAAQEAAG1hdHQJAAAAAQEAAERvY3VtZW50cwAAABAAAAABAQAAQWJvdXQgU3RhY2tzLnBkZhAAAAABBgAAEAAAACAAAAAsAAAAQAAAAAgAAAAEAwAAFmYAAAAAAAAIAAAABAMAAJTnBAAAAAAACAAAAAQDAAAzgQYAAAAAAAgAAAAEAwAAaIEGAAAAAAAQAAAAAQYAAHAAAACAAAAAkAAAAKAAAAAIAAAAAAQAAEGrrCMaAAAAGAAAAAECAAABAAAAAAAAAA8AAAAAAAAAAAAAAAAAAAAIAAAABAMAAAIAAAAAAAAABAAAAAMDAAD1AQAACAAAAAEJAABmaWxlOi8vLwwAAAABAQAATWFjaW50b3NoIEhECAAAAAQDAAAAkIKW5wAAAAgAAAAABAAAQcYvyAaAAAAkAAAAAQEAAEEyNjQwRDgyLTA0MjktNDhFQy04NTUyLUIyMDUyN0Q2ODE0NxgAAAABAgAAgQAAAAEAAADvEwAAAQAAAAAAAAAAAAAAAQAAAAEBAAAvAAAAAAAAAAEFAADMAAAA\/v\/\/\/wEAAAAAAAAAEAAAAAQQAABYAAAAAAAAAAUQAACwAAAAAAAAABAQAADYAAAAAAAAAEAQAADIAAAAAAAAAAIgAACkAQAAAAAAAAUgAAAUAQAAAAAAABAgAAAkAQAAAAAAABEgAABYAQAAAAAAABIgAAA4AQAAAAAAABMgAABIAQAAAAAAACAgAACEAQAAAAAAADAgAACwAQAAAAAAAAHAAAD4AAAAAAAAABHAAAAgAAAAAAAAABLAAAAIAQAAAAAAABDQAAAEAAAAAAAAAA==","id":"D64D6195-7645-4C1D-935F-3E34286534F5
```

```
Ym9va7wCAAAAAAQQMAAAAHpF1pJqGLcWddHXLQvPkeOfGMmuf3ETMWg0pWwJoWO0uAEAAAQAAAADAwAAABgAKAUAAAABAQAAVXNlcnMAAAAEAAAAAQEAAG1hdHQJAAAAAQEAAERvY3VtZW50cwAAABAAAAABAQAAQWJvdXQgU3RhY2tzLnBkZhAAAAABBgAAEAAAACAAAAAsAAAAQAAAAAgAAAAEAwAAFmYAAAAAAAAIAAAABAMAAJTnBAAAAAAACAAAAAQDAAAzgQYAAAAAAAgAAAAEAwAAaIEGAAAAAAAQAAAAAQYAAHAAAACAAAAAkAAAAKAAAAAIAAAAAAQAAEGrrCMaAAAAGAAAAAECAAABAAAAAAAAAA8AAAAAAAAAAAAAAAAAAAAIAAAABAMAAAIAAAAAAAAABAAAAAMDAAD1AQAACAAAAAEJAABmaWxlOi8vLwwAAAABAQAATWFjaW50b3NoIEhECAAAAAQDAAAAkIKW5wAAAAgAAAAABAAAQcYvyAaAAAAkAAAAAQEAAEEyNjQwRDgyLTA0MjktNDhFQy04NTUyLUIyMDUyN0Q2ODE0NxgAAAABAgAAgQAAAAEAAADvEwAAAQAAAAAAAAAAAAAAAQAAAAEBAAAvAAAAAAAAAAEFAADMAAAA/v///wEAAAAAAAAAEAAAAAQQAABYAAAAAAAAAAUQAACwAAAAAAAAABAQAADYAAAAAAAAAEAQAADIAAAAAAAAAAIgAACkAQAAAAAAAAUgAAAUAQAAAAAAABAgAAAkAQAAAAAAABEgAABYAQAAAAAAABIgAAA4AQAAAAAAABMgAABIAQAAAAAAACAgAACEAQAAAAAAADAgAACwAQAAAAAAAAHAAAD4AAAAAAAAABHAAAAgAAAAAAAAABLAAAAIAQAAAAAAABDQAAAEAAAAAAAAAA==
```

<img width="712" alt="Screen shot 2024-09-25 at 19 33 09" src="https://github.com/user-attachments/assets/68cd18ff-4870-48f3-a50e-aaf52a40edc2">

<img width="886" alt="image" src="https://github.com/user-attachments/assets/3bbc8054-d346-4b25-93a6-633fee87f1c8">

<img width="1006" alt="Screen shot 2024-09-25 at 19 32 33" src="https://github.com/user-attachments/assets/f9e46d83-9ca3-4df2-b09b-45917df5d178">

**Comment by runxel** [comment]
Oh yeah, thank you for the detailed explanation. I figured it out already on the way 😄.
My comment still holds up. Precize and Stapler are choking on it.

**Comment by gingerbeardman** [comment]
I'm not really sure what you're trying to do?

If Precize does not process it, you've done something wrong.

Not sure what you mean about Stapler? You should not be modifying the Stapler file manually. I have not proposed you should make any changes there. How is it "choking". 

If you want to share a working Stapler file with missing items I can look into it. Keep in mind it might have personal details in the path names.

Basically, I need much much more information to do anything about this one.

**Comment by runxel** [comment]
Sorry Matt if I wasn't precise enough.
What I meant was, that Stapler couldn't even open it's own .stapled file, but instead froze and not recover.
Then I tried to put in the decoded JSON part with the real content only, everything else stripped off, into Precize, which **crashed** on that input.
The screenshot above is just from a random Base64 decoder, beacuse I wanted to see if I can spot any issues like this. But then I saw that even from working stapler files the content "dumbly" decoded looks like garbled borf, so nothing to see there.

Today tho, after a restart, did 2 of 3 items recover and they are available again (= not "unknown"), also the stapler file itself opens again normally. It really is super mysterious.

**Comment by runxel** [comment]
That's what I thought. It can not really be an issue in Stapler per se, if the content can got kaputt or be recovered at random.
I'll keep an eye on it – but I only had trouble with the NAS so far, which means it has something to do with how macOS accesses that. Probably not even much we can do here.

**Comment by gingerbeardman** [comment]
Finally! 😅

Done in 1.3 MAS, releasing ASAP
